pontoon/sync/formats/l20nformat.py

```diff
@@ -0,0 +1,107 @@
+import codecs
+
+from l20n.format.parser import L20nParser
+from pontoon.sync.formats.base import ParsedResource
+from pontoon.sync.vcs.models import VCSTranslation
+
+
+class L20NStringEntity(VCSTranslation):
+    def __init__(self, entry, translation_string, comments, order):
+        super(L20NStringEntity, self).__init__(
+            key=entry.id.name,
+            strings={None: translation_string},
+            comments=comments,
+            fuzzy=False,
+            order=order,
+        )
+
+    def __repr__(self):
+        return '<L20NStringEntity {key}>'.format(self.key.encode('utf-8'))
+
+
+class L20NHashEntity(VCSTranslation):
+    def __init__(self, entry, strings, comments, order):
+        super(L20NHashEntity, self).__init__(
+            key=entry.id.name,
+            strings=strings,
+            comments=comments,
+            fuzzy=False,
+            order=order,
+        )
+
+    def __repr__(self):
+        return '<L20NHashEntity {key}>'.format(self.key.encode('utf-8'))
+
+    def update_plurals(self, locale):
+        """
+        Replaces string keys with their proper cldr codes.
+        """
+        def map_plural(key):
+            idx = locale.__class__.cldr_plural_to_id(key)
+            return locale.cldr_plurals_list().index(idx)
+        self.strings = {map_plural(key):val for key, val in self.strings.items()}
+
+
+class L20NAttributeEntity(VCSTranslation):
+    def __init__(self, entry, attr, order):
+        super(L20NAttributeEntity, self).__init__(
+            key=".".join([entry.id.name, attr.id.name]),
+            strings={None: attr.value.source},
+            comments=None,
+            fuzzy=False,
+            order=order,
+        )
+
+
+class L20NResource(ParsedResource):
+    def __init__(self, source_path, structure):
+        self.source_path = source_path
+        self.structure = structure
+        self._translations = self.map_objects(structure.body)
+
+    @property
+    def translations(self):
+        return self._translations
+
+    def map_objects(self, entities):
+        """
+        Method maps objects from parser to the VCS translations.
+        """
+        comments = []
+        translations = []
+        order = 0
+
+        for obj in entities:
+            if obj.type == 'Entity':
+                if obj.value.type == 'String':
+                    translations.append(L20NStringEntity(obj, obj.value.source, comments, order))
+                    order += 1
+
+                elif obj.value.type == 'Hash':
+                    strings = {item.id.name:item.value.source for item in obj.value.items}
```

Comment:
You need to map CLDR values to integers, see:
https://github.com/mozilla/pontoon/pull/315/files#diff-c5dc7820ff79f0945fc21de86ffc85aaL515